### PR TITLE
[DEV-2314] observation_table: tag entity_ids when creating observation table

### PR DESCRIPTION
**Pull request by jevonyeoh**
## Description
Set the entity_ids on the observation table when we create it from a target, view or source_table.

We infer the entity IDs by looking up column names from the table, and adding the entity information onto the `ColumnSpec`.

## Type of Change

Label this pull request to place it under the correct category in Release Notes:

|        **Pull Request Label**         | **Category in Release Notes** |
|:-------------------------------------:|:-----------------------------:|
|       `enhancement`, `feature`        |          🚀 Features          |
| `bug`, `refactoring`, `bugfix`, `fix` |    🔧 Fixes & Refactoring     |
|       `build`, `ci`, `testing`        |    📦 Build System & CI/CD    |
|              `breaking`               |      💥 Breaking Changes      |
|            `documentation`            |       📝 Documentation        |
|            `dependencies`             |    ⬆️ Dependencies updates    |

## Checklist

- [ ] I have read the [`CODE_OF_CONDUCT.md`](https://github.com/featurebyte/featurebyte/blob/main/CODE_OF_CONDUCT.md) and [`CONTRIBUTING.md`](https://github.com/featurebyte/featurebyte/blob/main/CONTRIBUTING.md) guides.
- [ ] I have written tests for the changes made.
- [ ] I have written docstrings in [NumpyDoc format](https://numpydoc.readthedocs.io/en/latest/format.html#docstring-standard)
- [ ] I have labeled my Pull Request correctly

### CODE REVIEW

File: featurebyte/schema/observation_table.py [hunk 1]
```diff
@@ -20,6 +20,7 @@ class ObservationTableCreate(BaseRequestTableCreate):
 
     sample_rows: Optional[int] = Field(ge=0)
     request_input: ObservationInput
+    entity_ids: List[PydanticObjectId] = Field(default_factory=list)
```
Comment (gee-senbong) [inline featurebyte/schema/observation_table.py:23]: Curious do we need to link the entity to the column?

Comment (jevonyeoh) [reply featurebyte/schema/observation_table.py:23]: hm yea i think that makes sense. let me explore / update!

File: featurebyte/service/materialized_table.py [hunk 2]
```diff
@@ -95,10 +100,36 @@ async def generate_materialized_table_location(
...
+        serving_names_remapping = {} if serving_names_remapping is None else serving_names_remapping
+        serving_names_reverse_lookup = {
+            value: key for key, value in serving_names_remapping.items()
+        }
+
+        # Build mapping of entity serving name to entity id
+        entity_serving_name_to_entity_id = {}
+        async for entity in self.entity_service.list_documents_iterator(query_filter={}):
+            for serving_name in entity.serving_names:
+                entity_serving_name_to_entity_id[serving_name] = entity.id
```
Comment (gee-senbong) [inline featurebyte/service/materialized_table.py:115]: Should we use `serving_names_remapping` here?

File: tests/unit/service/test_observation_table.py [hunk 3]
```diff
@@ -155,8 +157,12 @@ async def test_validate__most_recent_point_in_time(
 
     assert metadata == {
         "columns_info": [
-            {"name": "POINT_IN_TIME", "dtype": "TIMESTAMP"},
-            {"name": "cust_id", "dtype": "VARCHAR"},
+            {"name": "POINT_IN_TIME", "dtype": "TIMESTAMP", "entity_id": None},
+            {
+                "name": "cust_id",
+                "dtype": "VARCHAR",
+                "entity_id": ObjectId("63f94ed6ea1f050131379214"),
```
Comment (gee-senbong) [inline tests/unit/service/test_observation_table.py:164]: nit: Would be good to use something like `cust_id_entity.id` or `metadata["columns_info"][1]["entity_id"]` here.

**Review by gee-senbong — APPROVED** [review]
LGTM

**Pull request merged**